src/specular/stages/batches.rs

```diff
@@ -177,60 +186,73 @@ where
     }
 }
 
-/// Decodes a batch depending on its version.
-/// TODO: consider returning a generic/trait-type to support multiple versions.
+/// Decode Specular batches from a [SpecularBatcherTransaction] based on its version.
+/// Currently only version 0 is supported.
+// TODO: consider returning a generic/trait-type to support multiple versions.
 fn decode_batches(
     batcher_tx: &SpecularBatcherTransaction,
     state: &RwLock<State>,
-    blocktime: u64,
+    config: &Config,
```

Comment:
Passing full config again. Given `blocktime` is what is needed, why not make this just that? Does it have to do with use of `Arc` and atomicity?

Reply:
Ah, I see where the oracle is needed in a deeper function